go/libkb/gpg_cli.go

```diff
@@ -170,6 +170,35 @@ func (g *GpgCLI) ExportKey(k PGPKeyBundle) (err error) {
 	return PickFirstError(e1, e2, e3)
 }
 
+func (g *GpgCLI) Sign(fp PGPFingerprint, payload []byte) (string, error) {
+	arg := RunGpg2Arg{
+		Arguments: []string{"--armor", "--sign", "--default-key", fp.String()},
```

Comment:
I would recommend -u since I believe that is what the node client uses and
therefore is tested across a bunch of platforms and versions.


Otherwise good to ship!

On Thursday, November 19, 2015, Patrick Crosby <notifications@github.com>
wrote:

> In go/libkb/gpg_cli.go
> <https://github.com/keybase/client/pull/1359#discussion_r45388463>:
>
> > @@ -170,6 +170,35 @@ func (g *GpgCLI) ExportKey(k PGPKeyBundle) (err error) {
> >  	return PickFirstError(e1, e2, e3)
> >  }
> >
> > +func (g *GpgCLI) Sign(fp PGPFingerprint, payload []byte) (string, error) {
> > +	arg := RunGpg2Arg{
> > +		Arguments: []string{"--armor", "--sign", "--default-key", fp.String()},
>
> They seem to be the same:
>
> -u, --local-user name
> Use name as the user ID to sign. This option is silently ignored for the list commands, so that it can be used in an options file.
>
> --default-key name
> Use name as default user ID for signatures. If this is not used the default user ID is the first user ID found in the secret keyring.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/keybase/client/pull/1359/files#r45388463>.
>